### ISSUE TITLE
No search on Power key (ie. power plant, substation, etc.)

### ISSUE THREAD

**Issue by pierzen** [twain47/Nominatim#523]
search for Objects with key=plant, substation, etc returns no result unless an other key is added such as building, man_made, etc.

Please add power key in the search list.

**Comment by dkewley** [comment]
Specific examples of existing power=* objects which I cannot find by name using Nominatum:

power=line
ref="San Onofre - Santiago"
other k=v also
http://www.openstreetmap.org/way/13282179

power=substation
name="Moulton Substation"
other k=v also
http://www.openstreetmap.org/way/61878182#map=18/33.60933/-117.69707

Neither is searchable by name/ref in Nominatum. How can they be made searchable?

**Comment by lonvia** [comment]
Sorry to say but that is such a special-interest application that it is unlikely that it will ever make it into the generic search index. The only exception might be power=plant.

**Comment by pierzen** [comment]
The power tag in OSM describes various important features.  For generating stations, we either see power=plant or power=generatorThere are usages of OSM and Nominatim other then to spot a commerce or street on a phone. And we have to be carefull to offer an equilibrium and service these various aspects. The Power tag is an important OSM feature and contributors from various countries are tracing these features. Plants, Generators, Transmission substations and major power line (high voltage) are of interest for technical, political, environment aspects. 

We find relations and ways describing either Plants or Generators with "generator:source" that indicates the various generator sources (ie. solar, wind, photovoltaic, hydro, coal, gas, oil, nuclear, biomass, geothermal). 

Taginfo let us see the usage of some of these tags. 
power=plantpower-generator        295substation=transmission            10,800power=line + voltage>300,000  +40,500
voltage (power=line)315,000    208
330,000   2194345          249 +104
380,000    792 +163 +135
400,000 15810 +162 +150 +118 +107 +91 +74
420    190
440         193
500,000  8115 +63
735        129
750,000    214
765,000   713800        111

We can make the distinction between production, transmission and distribution. While distribution can be omitted, other aspects are important to follow. Medias or public in general should have the possibility to search for these features. OSM has the great advantage over other maps to let represents these features at various levels. Overpass let's evaluate how important are the power tags in the OSM Database. Searching only for Power relations, it returns 400,000 lines.  
Power production and Power transmission are tagged on ways and relations with power=Plant, power=generator substation=transmission power=line+voltage>100,000 (high voltage transmission lines).
Look at these relations
Solar Plant generators
http://www.openstreetmap.org/relation/6600903
Photovoltaic and Wind Farms plants, reported either as Plant or Generator
http://www.openstreetmap.org/relation/6004071#map=14/48.5248/-2.4903
http://www.openstreetmap.org/relation/368946#map=10/42.0916/-93.1723
http://www.openstreetmap.org/relation/1271386http://www.openstreetmap.org/relation/6047143
High Voltage Transmission Substationshttp://www.openstreetmap.org/relation/6277789http://www.openstreetmap.org/relation/6287658#map=15/55.9790/37.9699http://www.openstreetmap.org/way/33144100



High-Voltage Transmission Networkhttp://www.openstreetmap.org/relation/4504531#map=5/50.191/-69.038
High-Voltage HDVC Power Transmission Lineshttp://www.openstreetmap.org/relation/3392247
Power Generators, hydro-electric power plantshttp://www.openstreetmap.org/relation/1059409
Substation transmission + converter (HDVC)http://www.openstreetmap.org/relation/6034120
 
Pierre 


      De : Sarah Hoffmann <notifications@github.com>
 À : twain47/Nominatim <Nominatim@noreply.github.com> 
Cc : Pierre Beland <infosbelas-am@yahoo.fr>; Author <author@noreply.github.com>
 Envoyé le : mardi 27 Septembre 2016 17h20
 Objet : Re: [twain47/Nominatim] Objects with Power key (ie. power plant, substation, etc.) are not reported by Nominatim (#523)


  
Sorry to say but that is such a special-interest application that it is unlikely that it will ever make it into the generic search index. The only exception might be power=plant.—

**Comment by lonvia** [comment]
I had looked at the examples that have been provided in this ticket and the situation with name and ref tag is about as bad as for railways and hiking routes. See my comments about that [here](https://github.com/openstreetmap/Nominatim/issues/413#issuecomment-207590647). We need better tokenization before object with `power=*` can be included.

@pierzen deliberately mistagging objects to get them into Nominatim is not acceptable and certainly does not help to make your case here.

**Comment by gustavecha** [comment]
> @pierzen deliberately mistagging objects to get them into Nominatim is not acceptable and certainly does not help to make your case here.

I agree with you, maybe because i'm a little bit aware of how things go on OSM.

Earlier today, a new contributor told me he prefer using the name=* key instead of any ref:* subtags because name shows up in nominatim results.
It's not about mandatory rules or authority but of what people are encouraged to do to get what they want. Tools have to be flexible a bit as to not make tagging for render or tagging for geocoding more useful than consistent tagging.

**Comment by sinanna** [comment]
+1 to @pierzen and @gustavecha 

A bunch of mappers maps have resorted to tag for nomination because of this issue.

I've had e.g. [this](https://www.openstreetmap.org/changeset/58643842) discussion.

**Comment by drkludge** [comment]
@lonvia I do not understand the substation name embargo. I tried searching for [Cheatham Substation](https://www.openstreetmap.org/way/158143549) after the data was entered into the OSM.  In this case Cheatham Substation is named after the "Cheatham Farms" farm that used to be in the area until it was subdivided for housing and other commercial features.  Cheatham Farms is just a name.  I don't feel like I am doing anything wrong or trying to map for nominatim.  Currently, nominatim returns a location in Cheatham Tennessee.  That certainly is not correct for the substation but returns the Substation road in the Cheatham County. And clearly you match [Googles same results](https://www.google.com/search?ei=uV2dW9jLO4jb8AOm2YrYAg&q=%09Cheatham+substation&oq=%09Cheatham+substation&gs_l=psy-ab.3...2664.12122..12366...1.0..0.183.2138.12j9......0....1j2..gws-wiz.......0j0i71j0i22i30j0i22i10i30j33i160.AQ4V9bYtJmQ) but I was hoping for OSM nominatim to be better than Google in this case. I would hope the find my [Meadowbrook Substation in Phoenix AZ](https://www.openstreetmap.org/way/127703364) and not the [Meadowbrook Substation in Winchester,VA .](http://realtimeutilityengineers.com/project-profiles/substation-project-summaries/meadowbrook-substation/)  That is a useful feature of OSM because utility companies are loath to provide this kind-of information after 911.
![image](https://user-images.githubusercontent.com/1669057/45590151-fe1b7000-b8e6-11e8-8202-60357128c7fd.png)

**Comment by HellMapGoesCoding** [comment]
Here is an example of a solar power plant I was looking for with the exact name "Cēsu SES": https://www.openstreetmap.org/way/1318446434 . Looking through issues, I came across this one. Please make solar/power plants searchable by name. I do not see how this is a "special-interest application", this is a major installation of general interest to many people. I really don't want to be adding spurious tags to such features just so they appear in search results.
